php/class-ad-layers-widget.php

```diff
@@ -47,7 +47,7 @@ public function widget( $args, $instance ) {
 			}
 
 			// Display the ad unit
-			echo wp_kses_post( $args['before_widget'] . $ad_unit_html . $args['after_widget'] );
+			echo $args['before_widget'] . $ad_unit_html . $args['after_widget']; // WPCS: XSS okay.
```

Comment:
Random thought, but Instead of removing the wp_kses_post() completely could you pass an `$allowed_html` array wrapped in a filter?

e.g.

```
echo wp_kses_post(
	$args['before_widget'] . $ad_unit_html . $args['after_widget']
	apply_filters( 'ad_layers_widget_allowed_html', wp_kses_allowed_html() )
);
```

Could also pass a custom array instead of `wp_kses_allowed_html()` function.

Reply:
I don't see a strong case for this here since we already display the contents of `$ad_unit_html` elsewhere. Maybe I could `wp_kses_post` the contents of before/after widget instead.